core/src/impl/Kokkos_Half_MathematicalFunctions.hpp

```diff
@@ -427,7 +427,18 @@ KOKKOS_INLINE_FUNCTION Kokkos::Experimental::bhalf_t nextafter(Kokkos::Experimen
 #endif  // !(defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOS_COMPILER_MSVC))
 
 // isnormal
-KOKKOS_IMPL_MATH_HALF_FUNC_WRAPPER(KOKKOS_IMPL_MATH_UNARY_PREDICATE_HALF, signbit)
+#if defined(KOKKOS_HALF_T_IS_FLOAT) && !KOKKOS_HALF_T_IS_FLOAT
+KOKKOS_INLINE_FUNCTION bool signbit(Kokkos::Experimental::half_t x) {
+  constexpr std::uint16_t sign_mask = 0x8000;
```

Comment:
```suggestion
  constexpr std::uint16_t sign_mask = 1u<<15;
```
is much clearer fo me. I don't have an intuition was `0x8000` is in this context.